hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetRecordHandler.java

```diff
@@ -58,13 +70,73 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
     @Override
     public void validate(@NonNull final QueryContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+        final var op = context.query().transactionGetRecordOrThrow();
+
+        // The transaction ID must be specified
+        if (!op.hasTransactionID()) throw new PreCheckException(INVALID_TRANSACTION_ID);
+
+        // The record must exist for that transaction ID
+        final var txnId = op.transactionIDOrThrow();
+
+        // verify that the account id exist and not default
+        final var accountID = txnId.accountID();
+        if (accountID == null || accountID.equals(AccountID.DEFAULT)) {
+            throw new PreCheckException(INVALID_ACCOUNT_ID);
+        }
+
+        final var recordCache = context.createStore(RecordCache.class);
+        final var record = recordCache.getReceipt(txnId);
+        mustExist(record, INVALID_TRANSACTION_ID);
     }
 
     @Override
     public Response findResponse(@NonNull final QueryContext context, @NonNull final ResponseHeader header) {
         requireNonNull(context);
         requireNonNull(header);
-        throw new UnsupportedOperationException("Not implemented");
+        final var query = context.query();
+        final var op = query.transactionGetRecordOrThrow();
```

Comment:
Also here, should we be throwing a HandleException (or similar) instead of NPE?
Come to think of it, I wonder if we could significantly improve a lot of code if the `zzzOrThrow()` methods all optionally took an Exception to throw, and threw that, so we could pass in the correct (PreCheck/Handle/Whatever) exception.

Reply:
that's what we are doing in all other queries all over the place and I feel if we need / want to change it it need to be across the board.

Reply:
That's reasonable, still something we should consider in the near future.